### ISSUE TITLE
Opinion on Python in Google Kickstarter 

### ISSUE THREAD

**Issue by RohitSingh107** [kamyu104/GoogleKickStart-2022#1]
Sorry bro If I disturbed you.

I wanted to ask some questions.

I am new in competitive programming. I am comfortable in both C++ and Python. I love Python due to it's simplicity and clean syntax but is considered slow for competitive programming. Since you have completed many Google Kickstarts  in Python I believe you are right person to answer this questions, 

"is Python language suitable for Google Kickstart?", 
"Is Python too slow for Google Kickstart and competitive programming?",
 "How often you get TLE in Python/PyPy?"
"And Is there any other disadvantage in using Python?"

And finally would you "recommend" python for Google Kickstart?

Please take your time to reply, I am not in hurry.

Thank you 😊

**Comment by Skiii-16** [comment]
Hello bro,

Both CPP and python are no doubt good for coding but if you ask me I will
always prefer CPP over python because in python we can easily escape some
of the steps by using the libraries which in case of CPP one have to think
optimally and then have to implement it with keeping the TLE in mind which
helps in growth of both mind and your approach toward problem solving.
But in the end after all it's your decision of selecting a particular
language for coding as if you ask me I will recommend you to shift to CPP
as it will always helps you from competitive to yours academics(in
placement of colleges).

Wishing you all the best with your new journey and welcoming you to
competitive programming.

**Comment by kamyu104** [comment]
- For the most advanced competitive programming competitions, I prefer C++ over Python for the following reasons:
    - There are many optimized templates/libraries written in C++. You can use an out-of-the-box solution.
    - Some difficult/complex problems can only be solved in C++ at advanced rounds. In Python this is usually `TLE/MLE`.
    - Given the same time complexity solution, I have to do a lot of optimizations in Python to avoid `TLE/MLE`, but I do very little in C++.
- For `Google KickStart`, I prefer Python to C++ for the following reasons:
    - Rounds are for beginners. Almost any problem can be solved with Python.
    - I had to challenge myself to come up with a solution with better time complexity without the advantages of C++ language.
 - In conclusion:
      - Beginner rounds like `Leetcode / Qualification / KickStart / Round1`: I prefer `Python`.
      - Advanced rounds like `Round2 / Round3 / Finals`: I prefer `C++`.
